Fix wxID_SAVEAS label to meet design guidelines

dhowland opened the pull request:
Microsoft design guidelines for labels specify the following:

"Indicate a command that needs additional information (including confirmation) by adding an ellipsis at the end of the button label."

Refer to this document:  https://msdn.microsoft.com/en-us/library/windows/desktop/dn742402(v=vs.85).aspx

The wxID_SAVEAS stock item label does not conform.  Note that the documentation in docs/doxygen/mainpages/const_stockitems.h is already correct.  This change fixes the code such that it matches the documentation.  The bug was introduced when the label was updated incorrectly in Aug 2011.

dhowland committed:
Fix wxID_SAVEAS label to meet design guidelines

This bug was introduced way back in Aug 2011 by
73d1308b85df21aa1d69b74a37c4f88c53fe661f

This fixes the code to match the exsiting documentation in
docs/doxygen/mainpages/const_stockitems.h

adrientetar commented on src/common/stockitem.cpp:
```diff
@@ -188,7 +188,7 @@ wxString wxGetStockLabel(wxWindowID id, long flags)
         STOCKITEM(wxID_REVERT_TO_SAVED,     _("Revert to Saved"),     _("Revert to Saved"));
         STOCKITEM(wxID_SAVE,                _("&Save"),               _("Save"));
-        STOCKITEM(wxID_SAVEAS,              _("&Save as"),            _("Save as"));
+        STOCKITEM(wxID_SAVEAS,              _("Save &as..."),         _("Save as..."));
```
While you're at it, could you capitalize the As? If you look at e.g. notepad.exe the As starts with a capital.

dhowland replied:
I left it lower case because the Microsoft document I linked used lower case for "Save as...", however I concede that "Save As..." looks better to me.  I checked and Word, Notepad, and Visual Studio all use upper case 'A' so I will change it.

While we're at it, Notepad has "Find..." and "Replace..."

catalinr replied:
AFAIR the trailing dots tell the user that other action will be required after choosing that option. It's more like a UI rule than a native or not thing, isn't it? Also, it has nothing to do with the modality of the dialog (see Notepad++), or even the presence of a dialog as is the case of Chrome.

As `Find` and `Replace` will probably always show a search box, they should most likely get the trailing dots, it's just the reason for it that I thought of pointing out.

AliKet replied:
Sublime Text, Geany, NetBeans, gedit and LibreOffice all have these entries:
Find...
[Find and] Replace... 
Save As...

dhowland committed:
Use more aesthetic "Save As..."

dhowland committed:
Bring stock Find and Replace in line with standards

Find and Replace would typically require user input to receive the
search string from the user, therefore their labels should end with an
elipsis.  Also, these functions operate on documents, not selections.
Finally, Ctrl-H is the commonly used shortcut for Replace, rather than
Help (which is usually F1)

dhowland committed:
Fill out help strings for menu items

dhowland commented:
The more I looked, the more "abnormal" stuff I found.  I think the edits I've put in so far bring it to a much better place.  At least, it covers the application I'm working on.

vadz commented:
The changes look good to me, thanks! I'll merge them soon.
